cvat/apps/quality_control/signals.py

```diff
@@ -37,8 +38,15 @@ def __save_job__update_quality_metrics(instance, created, **kwargs):
     else:
         assert False
 
-    for task in tasks:
-        qc.QualityReportUpdateManager().schedule_quality_autoupdate_job(task)
+    def schedule_autoupdate_jobs():
+        for task in tasks:
+            if task.id is None:
+                # The task may have been deleted after the on_commit call.
+                continue
```

Comment:
Why are you sure that instance from the list will be actual after deleting the row in the database? I've checked that e.g. after updating a project with several tasks in one transaction and deleting one project task after closing the first transaction -> related task instance from the list has "outdated" properties. Probably you need to check directly that an object exists in db.

Reply:
My goal here isn't really to _ensure_ that the task exists in the database. It's not possible to do that anyhow, since we're not in a transaction, and another request could delete the task right after we check it.

Rather, I just want to make sure that `task.id` is not `None` (even if it's outdated), because `schedule_quality_autoupdate_job` references it. I'm not sure what'll happen if `schedule_quality_autoupdate_job` encounters an ID of `None`, and I can't assume that it will crash safely, so I'd rather prevent this from happening.